src/io/result_io.jl

```diff
@@ -383,13 +383,13 @@ function setup_result_store!(domain::Domain, scen_spec::DataFrame)::Tuple
     n_scenarios = nrow(scen_spec)
 
     # Set up stores for each metric
-    function dim_lengths(metric_structure)
+    function dim_lengths(metric_structure::NTuple{3,Symbol})
         dl = []
-        for d in metric_structure
+        for d in string.(metric_structure)
```

Comment:
Isn't it better to avoid casting to string and instead update the comparisons in the if/elseif chain to compare against Symbols?

e.g., `d == :timesteps`

Reply:
Done.